sql/parser/builtins.go

```diff
@@ -72,6 +75,74 @@ var builtins = map[string]builtin{
 	"upper": stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToUpper(s)), nil
 	}),
+
+	// TODO(XisiHuang): support the substring(str FROM x FOR y) syntax.
+	"substring": {
+		nArgs: -1,
+		fn: func(args DTuple) (Datum, error) {
+			argsNum := len(args)
+			if argsNum != 2 && argsNum != 3 {
+				return DNull, fmt.Errorf("incorrect number of arguments: %d vs %s",
+					argsNum, "2 or 3")
+			}
+
+			dstr, ok := args[0].(DString)
+			if !ok {
+				return DNull, argTypeError(args[0], dstr.Type())
+			}
+			str := string(dstr)
```

Comment:
I had a similar thought - why don't we use the full signature (name + slice of arg types) to identify each method?